### PR TITLE
python/raindrops: Update mentoring.md

### PR DESCRIPTION
**Pull request by vidhisshah**
The notes were a little confusing. 
Made some changes and did some reordering to improve readability

### CODE REVIEW

File: tracks/python/exercises/raindrops/mentoring.md [hunk 1]
```diff
@@ -26,12 +26,12 @@ def raindrops(val: int) -> str:
     return result
 ```
 
-#### Solutions That Need Work
+### Solutions That Need Work
```
Comment (SleeplessByte) [inline tracks/python/exercises/raindrops/mentoring.md]:
```suggestion
### Common Suggestions
```

It think this actually works in terms of this document :)

Comment (vidhisshah) [reply tracks/python/exercises/raindrops/mentoring.md]: "Solutions that need work" suggests to the mentor that the student's solution though correct, still needs improvement and optimization. 
"Common suggestions" seems more ambiguous. 
But I understand why you're saying that, and I think the Talking Points section covers that. :)

Comment (SleeplessByte) [reply tracks/python/exercises/raindrops/mentoring.md]: Yes, I'm just trying it to match the other notes in the track

**Comment by SleeplessByte** [comment]
@vidhisshah did you close this intentionally?

**Comment by SleeplessByte** [comment]
Thank you!

**Pull request merged**